File: rust/src/models/invoice_status.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(
    Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
)]
pub enum InvoiceStatus {
    #[default]
    #[serde(rename = "DRAFT")]
    Draft,

    #[serde(rename = "FINALIZED")]
    Finalized,

    #[serde(rename = "UNCOLLECTIBLE")]
    Uncollectible,

    #[serde(rename = "VOID")]
    Void,
}

impl fmt::Display for InvoiceStatus {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::Draft => "DRAFT",
            Self::Finalized => "FINALIZED",
            Self::Uncollectible => "UNCOLLECTIBLE",
            Self::Void => "VOID",
        };
        f.write_str(value)
    }
}

File: typescript/src/models/invoiceStatus.ts
// this file is @generated

export enum InvoiceStatus {
  Draft = "DRAFT",
  Finalized = "FINALIZED",
  Uncollectible = "UNCOLLECTIBLE",
  Void = "VOID",
  Closed = "CLOSED",
}

export const InvoiceStatusSerializer = {
  _fromJsonObject(object: any): InvoiceStatus {
    return object;
  },

  _toJsonObject(self: InvoiceStatus): any {
    return self;
  },
};
